src/frontends/tensorflow_common/src/op/adjust_saturation.cpp

```diff
@@ -0,0 +1,202 @@
+            hue_case_b   // Use hue_case_b otherwise (b is max)
+        )
+    );
+
+    // range = 0.0: hh = 0
+    auto hh_zero_range = make_shared<v1::Select>(make_shared<v1::Equal>(range, const_zero_f_), const_zero_f_, hh);
+
+    // hh < 0.0: hh = hh + 1
+    auto hh_final = make_shared<v1::Select>(make_shared<v1::Less>(hh, const_zero_f_), make_shared<v1::Add>(hh_zero_range, const_one_f_), hh_zero_range);
+
+    return make_shared<tuple<shared_ptr<Node>, shared_ptr<Node>, shared_ptr<Node>>>(hh_final, ss, vv);
+}
+
+shared_ptr<Node> hsv_to_rgb(shared_ptr<Node> h, shared_ptr<Node> s, shared_ptr<Node> v, element::Type type) {
+    // image format conversion based on https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/image/adjust_saturation_op.cc
+    auto const_six_f_ = make_shared<v0::Constant>(type, Shape{}, 6.0f);
+    auto const_two_f_ = make_shared<v0::Constant>(type, Shape{}, 2.0f);
+    auto const_one_f_ = make_shared<v0::Constant>(type, Shape{}, 1.0f);
+
+    auto const_minus_one_i_ = make_shared<v0::Constant>(element::i32, Shape{}, -1);
+    auto const_minus_two_i_ = make_shared<v0::Constant>(element::i32, Shape{}, -2);
+    
+    // c = s * v;
+    auto c = make_shared<v1::Multiply>(s, v);
+    // m = v - c;
+    auto m = make_shared<v1::Subtract>(v, c);
+    // dh = h * 6;
+    auto dh = make_shared<v1::Multiply>(h, const_six_f_);
+
+    // fmodu rounded to within [0, 2)
+    auto fmodu = make_shared<v1::FloorMod>(dh, const_two_f_);
+
+    //  x = c * (1 - std::abs(fmodu - 1));
+    auto x = make_shared<v1::Multiply>(c, make_shared<v1::Subtract>(const_one_f_, make_shared<v0::Abs>(make_shared<v1::Subtract>(fmodu, const_one_f_))));
+
+    // h_category: [batch_dims..., H, W, 1]
+    auto h_category = make_shared<v0::Convert>(make_shared<v0::Floor>(dh), element::i32);
+
+    auto zeros = make_shared<v0::Constant>(type, x->get_shape(), 0.0f);
+
+    auto rr_options = NodeVector{c, x, zeros, zeros, x, c};
+    auto gg_options = NodeVector{x, c, c, x, zeros, zeros};
+    auto bb_options = NodeVector{zeros, zeros, x, c, c, x};
+
+    // rr_concat: [batch_dims..., H, W, 6]
+    auto rr_concat = make_shared<v0::Concat>(rr_options, -1); 
+    auto gg_concat = make_shared<v0::Concat>(gg_options, -1);
+    auto bb_concat = make_shared<v0::Concat>(bb_options, -1);
+
+    // rr_unsqueeze: [batch_dims..., H, W, 6, 1]
+    auto rr_unsqueeze = make_shared<v0::Unsqueeze>(rr_concat, const_minus_one_i_);
+    auto gg_unsqueeze = make_shared<v0::Unsqueeze>(gg_concat, const_minus_one_i_);
+    auto bb_unsqueeze = make_shared<v0::Unsqueeze>(bb_concat, const_minus_one_i_);
+
+    // rgb_options: [batch_dims..., H, W, 6, 3]
+    auto rgb_options = make_shared<v0::Concat>(NodeVector{rr_unsqueeze, gg_unsqueeze, bb_unsqueeze}, -1);
+
+    // use a gather operation to select the correct channel values based on h_category
+    // rgb: [batch_dims..., H, W, 3]
+    int batch_dim = rgb_options->get_shape().size() - 2;
```

Comment:
can we use `batch_dim = -3`? please try because `get_shape` throws exception in case partial or dynamic shapes that is quite frequent case

Reply:
It operates correctly with batch_dim = -1. -2, -3 add more axes to the var. I am still not very clear about the use of batch_dim in `Gather` though.